modules/deployment-model/src/main/java/org/mule/runtime/deployment/model/internal/plugin/BundlePluginDependenciesResolver.java

```diff
@@ -92,6 +102,29 @@ public BundlePluginDependenciesResolver(ArtifactDescriptorFactory<ArtifactPlugin
     return resolvedPlugins;
   }
 
+  private void verifyPluginExportedPackages(List<ArtifactPluginDescriptor> plugins) {
+    final Map<String, List<String>> exportedPackages = new HashMap<>();
+
+    boolean error = false;
+    for (ArtifactPluginDescriptor plugin : plugins) {
+      for (String packageName : plugin.getClassLoaderModel().getExportedPackages()) {
+        List<String> exportedOn = exportedPackages.get(packageName);
+
+        if (exportedOn == null) {
+          exportedOn = new LinkedList<>();
+          exportedPackages.put(packageName, exportedOn);
+        } else {
+          error = true;
+        }
+        exportedOn.add(plugin.getName());
+      }
+    }
+
+    if (error) {
+      throw new DuplicateExportedPackageException(exportedPackages);
```

Comment:
I believe until [MULE-11464](https://www.mulesoft.org/jira/browse/MULE-11464) is fixed, this PR will break the build as some connectors export other plugins' packages.
Such as the HTTP with Sockets, or anyone relying on the file-extension-common